### ISSUE TITLE
E2-4: Build mock submission factory for testing

### ISSUE THREAD

**Issue by jkbrooks** [jkbrooks/autoresearch_network#6]
## E2-4: Build mock submission factory for testing

**Parent Epic:** #2 — Protocol Definition  
**Depends on:** #3 (ExperimentSubmission), #4 (HardwareTier + constants), #5 (validation)

---

### What This Is

A factory class that generates realistic synthetic `ExperimentSubmission` instances for use in tests, demos, and downstream epic development (miner/validator). This eliminates the need for a real GPU or Bittensor chain to test any protocol logic.

### Where It Goes

`autoresearch/mock.py`

### Interface

```python
from autoresearch.mock import MockSubmissionFactory
from autoresearch.constants import HardwareTier

factory = MockSubmissionFactory(seed=42)

# Generate a valid submission that improved on the baseline
good = factory.make_submission(
    baseline_val_bpb=0.997,
    tier=HardwareTier.LARGE,
    improvement=0.003,  # new val_bpb will be 0.994
)

# Generate a valid submission that did NOT improve
neutral = factory.make_submission(
    baseline_val_bpb=0.997,
    tier=HardwareTier.LARGE,
    improvement=-0.002,  # regression: new val_bpb will be 0.999
)

# Generate various invalid submissions for negative testing
bogus_bpb = factory.make_invalid_submission(reason="bogus_bpb")
missing_source = factory.make_invalid_submission(reason="missing_train_py")
identical_source = factory.make_invalid_submission(reason="identical_train_py")
impossible_improvement = factory.make_invalid_submission(reason="impossible_improvement")
wrong_tier = factory.make_invalid_submission(reason="invalid_tier")
```

### Implementation Details

**`make_submission()`:**
- Generates a realistic `ExperimentSubmission` with all validator fields + miner response fields populated.
- `baseline_train_py` should be a short but structurally valid Python snippet (not the full 600-line `train.py` — use a ~20-line synthetic training script for test payloads).
- `train_py` should be the baseline with a small modification (e.g., a changed hyperparameter comment or constant) — must differ from baseline but be valid Python.
- `elapsed_wall_seconds` should be a realistic value for the tier (e.g., 290–310 for a 5-minute run).
- `peak_vram_mb` should fall within the tier's `TIER_PLAUSIBILITY` range.
- `run_log_tail` should contain a realistic metrics summary block matching the format from https://github.com/karpathy/autoresearch/blob/master/program.md#output-format.
- Use the `seed` parameter for deterministic output (seeded `random.Random` instance).

**`make_invalid_submission(reason=...)`:**
- Each `reason` string maps to a specific validation rule violation from #5:
  - `"bogus_bpb"` → `val_bpb = 0.1` (below MIN_PLAUSIBLE_VAL_BPB)
  - `"missing_train_py"` → `train_py = None`
  - `"identical_train_py"` → `train_py == baseline_train_py`
  - `"impossible_improvement"` → improvement > MAX_SINGLE_STEP_IMPROVEMENT
  - `"invalid_tier"` → `hardware_tier = "quantum"`
  - `"elapsed_too_short"` → `elapsed_wall_seconds = 10`
  - `"elapsed_too_long"` → `elapsed_wall_seconds = 1200`
- Unknown reason strings raise `ValueError`.

### Acceptance Criteria

- [ ] `MockSubmissionFactory` exists in `autoresearch/mock.py`
- [ ] `make_submission()` returns an `ExperimentSubmission` that passes `validate()` by default
- [ ] `make_submission(improvement=<positive>)` produces `val_bpb < baseline_val_bpb`
- [ ] `make_submission(improvement=<negative>)` produces `val_bpb > baseline_val_bpb`
- [ ] Each `make_invalid_submission(reason=...)` variant fails `validate()` with the expected `ValueError`
- [ ] Deterministic: same seed produces identical output
- [ ] No GPU, chain, or network access required

### Test Expectations

In `tests/test_protocol.py`:

- `test_mock_factory_valid_submission_passes_validation`
- `test_mock_factory_improvement_math` — positive improvement → lower bpb, negative → higher
- `test_mock_factory_deterministic` — same seed → same output
- `test_mock_factory_invalid_bogus_bpb_fails_validation`
- `test_mock_factory_invalid_missing_train_py_fails_validation`
- `test_mock_factory_invalid_identical_train_py_fails_validation`
- `test_mock_factory_invalid_impossible_improvement_fails_validation`
- `test_mock_factory_invalid_tier_fails_validation`
- `test_mock_factory_unknown_reason_raises_value_error`
